streaming/src/main/scala/org/apache/spark/streaming/dstream/SocketInputDStream.scala

```diff
@@ -51,29 +51,44 @@ class SocketReceiver[T: ClassTag](
     storageLevel: StorageLevel
   ) extends Receiver[T](storageLevel) with Logging {
 
+  private var socket: Socket = _
+
   def onStart() {
-    // Start the thread that receives data over a connection
-    new Thread("Socket Receiver") {
-      setDaemon(true)
-      override def run() { receive() }
-    }.start()
+    try {
+      logInfo(s"Connecting to $host:$port")
+      socket = new Socket(host, port)
+      logInfo(s"Connected to $host:$port")
+    } catch {
+      case NonFatal(e) =>
```

Comment:
Hm, I suppose now you're restarting here on any non-fatal error in connecting, not just `ConnectException`. Maybe that's OK. It's simpler to return here rather than then have to check if the socket initialized. Also the info logs seem like they don't belong in the try-catch; they can't fail.